### ISSUE TITLE
SelectedNode Feature Like Blazorise

### ISSUE THREAD

**Issue by christophwille** [excubo-ag/Blazor.TreeViews#29]
I have seen #20. I want to second the request to have first-class support for single item selection. 

Scenario: I need to switch away from https://blazorise.com/docs/extensions/treeview/ for on-demand loading (I need a proper async solution). Having a SelectedNode-style similar to other TreeViews (eg RadzenTree) out there would be great.

**Comment by stefanloerwald** [comment]
Hi @christophwille,

from looking at the blazorise demo, I think what you want can be achieved quite simply with this library. Here's some code for you:

```html 
<TreeView Items="Items" GetParent="GetParent">
    <ItemTemplate>
        <div @onclick="() => { selectedItem = (selectedItem != context.Item) ? context.Item : null; }" 
             style="background-color: @(selectedItem == context.Item ? "Highlight" : "inherit")">
           @context.Item
        </div>
    </ItemTemplate>
</TreeView>
```

The onclick handler on an item will allow you set any field in your component. In this case, I just created a private field `selectedItem`. (I also choose to allow to deselect an item by clicking it again. That's achieved by the ternary expression.)

The style can be modified using this field as well.

I hope this helps. I've expanded the selection example in the demo. See here: https://excubo-ag.github.io/Blazor.TreeViews/selection

Does this solve your task?

BR
Stefan
